src/transport/SecureSessionMgr.cpp

```diff
@@ -161,14 +161,13 @@ CHIP_ERROR SecureSessionMgr::SendMessage(SecureSessionHandle session, PayloadHea
     VerifyOrExit(admin != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
     localNodeId = admin->GetNodeId();
 
-    if (payloadHeader.HasMessageType(Protocols::SecureChannel::MsgType::MsgCounterSyncReq) ||
```

Comment:
Why the changes here?  They don't seem related to this PR......

Reply:
This PR changes the SecureChannel protocol ID to 0, and this exposed a bug here. When calling `SendEncryptedMessage()` (defined above the function shown here), it default-constructs a `PayloadHeader`, which results in a `PayloadHeader` with `ProtocolID = 0` (SecureChannel) and `MessageID = 0` (MsgCounterSyncReq) and this was unintentionally causing the condition here to be true.

Reply:
I've requested @yufengwangca to review this because he recently edited this code.

Reply:
> which results in a PayloadHeader with ProtocolID = 0 (SecureChannel) and MessageID = 0 (MsgCounterSyncReq) 

Uh... we should probably have the default constructor use an invalid protocol id, or message id, or both!

Reply:
That's a great idea! I can add that to this PR if you think it's relevant enough

Reply:
Should we really want to use 0 as one of valid protocol ID?  maybe we should modify spec to avoid similar occident.

Reply:
We could try to change the spec, sure.  Might be easier to change the code, though.

@holbrookt Followup for the constructor change is ok, assuming @yufengwangca is OK with the change here.

Reply:
The other thing we could do is not pass random default-constructed things to functions that aren't supposed to use them and use Optional<> as needed instead.  Or a better API that doesn't involve constructing "headers" that are then unused.

Reply:
I am ok to solve this issue by putting default value in constructor for now. We already have default value kProtocol_NotSpecified defined.